server/src/main/java/io/crate/metadata/functions/Signature.java

```diff
@@ -327,6 +342,12 @@ public Signature(StreamInput in) throws IOException {
         int enumElements = in.readVInt();
         features = Collections.unmodifiableSet(EnumSets.unpackFromInt(enumElements, Scalar.Feature.class));
         bindingInfo = null;
+        Integer nullabilityOrdinal = in.readOptionalVInt();
+        if (nullabilityOrdinal == null) {
+            nullability = kind == FunctionType.SCALAR ? Scalar.Nullability.DEFAULT : null;
```

Comment:
why do we need `null` isn't `DEFAULT` enough? do we need a 4th value (null) for the `nullability`?

Reply:
Thank you for reviewing. I thought we could keep it as nulls for non scalar functions. `DEFAULT` should be ok too, would you prefer that?

Reply:
I think it can be confusing, I'd know, I tend to go with DEFAULT and not use another variant (null value).

Reply:
Thank you added a comment to clarify: https://github.com/crate/crate/pull/16133/commits/233b8b9f749efc9d5b0a735cf6d16fe5ea5b6ed8#diff-49ced1cc3c653bee02f981f5a97ec6920eb63824c4747fbdb9c6013c635474cfR233-R234